compiler/dyno/lib/uast/chpl-syntax-printer.cpp

```diff
@@ -0,0 +1,1065 @@
+        ss_ << " ";
+        printAst(ss_, node->actual(0));
+      } else {
+        if (node->callUsedSquareBrackets()) {
+          interpose(ss_, node->actuals(), ", ", "[", "]");
+        } else {
+          interpose(ss_, node->actuals(), ", ", "(", ")");
+        }
+      }
+
+    }
+
+    void visit(const For* node) {
+      ss_ << "for ";
+      if (node->isParam()) {
+        ss_ << "param ";
+      }
+      if (node->index()) {
+        printAst(ss_, node->index());
+        ss_ << " in ";
+      }
+      printAst(ss_, node->iterand());
+      ss_ << " ";
+      printBlockWithStyle(ss_, node->blockStyle(), node->stmts(), "do ");
+    }
+
+    void visit(const Forall* node) {
+      ss_ << "forall ";
+
+      if (node->index()) {
+        printAst(ss_, node->index());
+        ss_ << " in ";
+      }
+      printAst(ss_, node->iterand());
+      if (node->withClause()) {
+        ss_<< " ";
+        printAst(ss_, node->withClause());
+      }
+      ss_ << " ";
+      printBlockWithStyle(ss_, node->blockStyle(), node->stmts(), "do ");
+    }
+
+    void visit(const Foreach* node) {
+      ss_ << "foreach ";
+
+      if (node->index()) {
+        printAst(ss_, node->index());
+        ss_ << " in ";
+      }
+      printAst(ss_, node->iterand());
+      ss_ << " ";
+      printBlockWithStyle(ss_, node->blockStyle(), node->stmts(), "do ");
+    }
+
+    void visit(const Formal* node) {
+      if (node->intent() != Formal::DEFAULT_INTENT) {
+        ss_ << kindToString(node->intent()) << " ";
+      }
+      ss_ << node->name().str();
+      if (const Expression* te = node->typeExpression()) {
+        ss_ << ": ";
+        printAst(ss_, te);
+      }
+      if (const Expression* ie = node->initExpression()) {
+        ss_ << " = ";
+        printAst(ss_, ie);
+      }
+    }
+
+    void visit(const ForwardingDecl* node) {
+      ss_ << "forwarding ";
+      if (node->expr()) {
+        printAst(ss_, node->expr());
+      }
+    }
+
+    void visit(const Function* node) {
+
+      printLinkages(node);
+
+      if (node->visibility() != Function::Visibility::DEFAULT_VISIBILITY) {
+        ss_ << kindToString(node->visibility()) << " ";
+      }
+
+      if (node->isOverride()) {
+        ss_ << "override ";
+      } else if (node->isInline()) {
+        ss_ << "inline ";
+      }
+
+      // Function Name
+      ss_ << kindToString(node->kind()) << " " << node->name().str();
+
+      // Formals
+      int numThisFormal = node->thisFormal() ? 1 : 0;
+      int nFormals = node->numFormals() - numThisFormal;
+      if (nFormals == 0 && node->isParenless()) {
+        // pass
+      } else if (nFormals == 0) {
+        ss_ << "()";
+      } else {
+        auto it = node->formals();
+        interpose(ss_, it.begin() + numThisFormal, it.end(), ", ", "(", ")");
+      }
+
+      // Return type
+      if (const Expression* e = node->returnType()) {
+        ss_ << " : ";
+        printAst(ss_, e);
+      }
+
+      // Return Intent
```

Comment:
I thought that return intent goes before the above `:`?

Reply:
yes, I changed the order now